11-type-constructors/algo/src/ICtx.hs

```diff
@@ -0,0 +1,69 @@
+module ICtx where
+
+import Common
+import Internal
+
+type EAssump = (TID, Term)
+
+type TAssump = (TID, Typ)
+
+type HAssump = (HID, Typ)
+
+data Ctx
+  = Nil
+  | Ctx :⌢ TAssump
+  | Ctx :⌢⌢ HAssump
+  | Ctx :⌢⌢⌢ EAssump
+  deriving (Show)
+
+-- TODO: this is c/p from ECtx
```

Comment:
what is this TODO saying? that the external and internal languages are the same?

Reply:
yeah that the external and internal type level languages are the same, but the internal language should have kind casts